substrate/frame/multi-asset-bounties/src/lib.rs

```diff
@@ -0,0 +1,1875 @@
+		type BeneficiaryLookup: StaticLookup<Target = Self::Beneficiary>;
+
+		/// Minimum value for a bounty.
+		#[pallet::constant]
+		type BountyValueMinimum: Get<BalanceOf<Self, I>>;
+
+		/// Minimum value for a child-bounty.
+		#[pallet::constant]
+		type ChildBountyValueMinimum: Get<BalanceOf<Self, I>>;
+
+		/// Maximum number of child bounties that can be added to a parent bounty.
+		#[pallet::constant]
+		type MaxActiveChildBountyCount: Get<u32>;
+
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+
+		/// Converts an `AssetKind` into the funding source account/location.
+		///
+		/// Used when initiating funding and refund payments to and from a bounty.
+		type FundingSource: TryConvert<
+			Self::AssetKind,
+			<<Self as pallet::Config<I>>::Paymaster as PayWithSource>::Source,
+		>;
+
+		/// Converts a bounty index and `AssetKind` into its account/location.
+		///
+		/// Used when initiating the funding, refund, and payout payments to and from a bounty.
+		type BountySource: TryConvert<
+			(BountyIndex, Self::AssetKind),
+			<<Self as pallet::Config<I>>::Paymaster as PayWithSource>::Source,
+		>;
+
+		/// Converts a parent bounty index, child bounty index, and `AssetKind` into the
+		/// child-bounty account/location.
+		///
+		/// Used when initiating the funding, refund, and payout payments to and from a
+		/// child-bounty.
+		type ChildBountySource: TryConvert<
+			(BountyIndex, BountyIndex, Self::AssetKind),
+			<<Self as pallet::Config<I>>::Paymaster as PayWithSource>::Source,
+		>;
+
+		/// Type for processing payments of [`Self::AssetKind`] from a `Source` in favor of
+		/// [`Self::Beneficiary`].
+		type Paymaster: PayWithSource<
+			Balance = Self::Balance,
+			Source = Self::Beneficiary,
+			Beneficiary = Self::Beneficiary,
+			AssetKind = Self::AssetKind,
+		>;
+
+		/// Type for converting the balance of an [`Self::AssetKind`] to the balance of the native
+		/// asset, solely for the purpose of asserting the result against the maximum allowed spend
+		/// amount of the [`Self::SpendOrigin`].
+		///
+		/// The conversion from the native asset balance to the balance of an [`Self::AssetKind`] is
+		/// used in benchmarks to convert [`Self::BountyValueMinimum`] to the asset kind amount.
+		type BalanceConverter: ConversionFromAssetBalance<Self::Balance, Self::AssetKind, BalanceOf<Self, I>>
+			+ ConversionToAssetBalance<BalanceOf<Self, I>, Self::AssetKind, Self::Balance>;
+
+		/// The preimage provider used for child-/bounty metadata.
+		type Preimages: QueryPreimage<H = Self::Hashing> + StorePreimage;
+
+		/// Means of associating a cost with committing to the curator role, which is incurred by
+		/// the child-/bounty curator.
+		///
+		/// The footprint accounts for the child-/bounty value in the native asset (returned in the
+		/// `Success` type of [`Self::SpendOrigin`]). The cost taken from the curator `AccountId`
```

Comment:
SpendOrigin origin returns MAX, you should/pass the bounty amount converted with `BalanceConverter`